draft-ietf-lpwan-schc-over-lorawan.md

```diff
@@ -674,92 +688,94 @@ purposes but not SCHC needs.
 ~~~~
 {: #Fig-fragmentation-downlink-header-abort title='Receiver-Abort packet (following an All-1 SCHC Fragment with incorrect RCS).'}
 
+#### Retransmission timer {#downlink-retransmission-timer}
+
+Class A and Class B or Class C devices do not manage retransmissions and timers
+in the same way.
+
+##### Class A devices
+
+Class A devices can only receive in an RX slot following the transmission of an
+uplink.  Therefore the SCHC gateway cannot initiate communication to a Class A
+device.
+Class A devices MUST implement an heartbeat sending uplink on port
+FPortCommandControl with an empty payload.  It will create a downlink
+opportunity for the SCHC gateway to start a SCHC session or send the SCHC ACK
+REQ if the retransmission timer expires.  Timing is application specific,
+RECOMMENDED value is one heartbeat every 12h.
+
+The device replies with a SCHC ACK message to every single fragment received
+from the SCHC gateway:
+
+* FCN=0: All fragment but the last have an FCN=0 (because window size is 1).
+  Following it the device MUST transmit the SCHC ACK. It MUST transmit up to
+  MAX_ACK_REQUESTS ACK messages before aborting. In order to progress the
+  fragmentation datagram as quickly as possible, the device should immediately
+  transmit those ACK if no SCHC downlink have been received during RX1 and RX2
+  window. The ACK bitmap is 1 bit long and is always 1.
+
+* FCN=1: The last fragment of a datagram. If the RCS is correct, the device
+  SHALL transmit the ACK with the bit C=1. This message might be lost
+  therefore the SCHC gateway MUST request a retransmission of this ACK in the
+  next downlink when the retransmission timer expires.  The device MUST keep
+  this ACK message in memory until it receives a downlink, on SCHC FPortDown
+  different from an SCHC ACK REQ: it indicates that the SCHC gateway has
+  received the ACK message.
+
+The SCHC gateway implements an inactivity timer with a RECOMMENDED duration
+of 48 hours. For devices with very low transmission rates (example 1 packet a
+day in normal operation), that duration may be extended, but this is application
+specific.
+
+#### Class B or Class C devices
+
+Class B devices can receive in scheduled RX slots or in RX slots following the
+transmission of an uplink. Class C devices are almost in constant reception.
+For those classes there is no need of an heartbeat to open RX window.
+
+RECOMMENDED retransmission timer value:
+
+* Class B: 3 times the ping slot periodicity.
+* Class C: 30 seconds
+
+The RECOMMENDED transmission timer value is 12 hours for both Class B and Class
```

Comment:
this sentence is a bit confusing. is it referring the retransmission timer again?

Reply:
I think i see the point now, it is the inactivity timer (perhaps?)  Maybe the context from ln:[725](https://github.com/Acklio/schc-over-lorawan/pull/32/files/5ad91fb7e35b9d9a5ad4427d2535382f9e268f31#diff-f81621167255b169a4bf3c62049366edR725) can be combined and moved out of `Retransmission timer` heading to `Inactivity timer` or something similar.

Reply:
Thanks Arun, fixed in 139bc6c